### PR TITLE
Login

### PR DESCRIPTION
**Pull request by felire**
### Summary

I made a couple of things. I've created the Login screen with the Smart and Dumb pattern. I've put the state login into the Login component. Also, I've implemented the Smart and Dumb pattern in all the screen that need it. I've implemented the SwitchNavigator to switch into the App and the Login at the 
beginning of the App. And, I've fixed a little Bug on the ScrollView on the DetailBook Component. 
I know that these are many changes but i've forgotten to separate them in other PR, sorry.

### Gif

transition3

### CODE REVIEW

File: WBooks/src/app/screens/AuthLoading/layout.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { ActivityIndicator, StatusBar, View } from 'react-native';
+
+function AuthLoadingLayout() {
+  return (
+    <View>
```
Comment (santinucera) [inline WBooks/src/app/screens/AuthLoading/layout.js]: Use fragment here

File: WBooks/src/app/screens/BookDetail/index.js [hunk 2]
```diff
@@ -7,14 +7,16 @@ import styles from './styles';
...
   return (
-    <ScrollView style={styles.bookDetailContainer}>
-      <View style={styles.detailBoxContainer}>
-        <DetailBox />
-      </View>
-      <View style={styles.commentBoxContainer}>
-        <CommentBox />
-      </View>
-    </ScrollView>
+    <View style={styles.bookDetailContainer}>
```
Comment (santinucera) [inline WBooks/src/app/screens/BookDetail/index.js]: Is really necessary this `View`?

File: WBooks/src/app/screens/Login/layout.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+function LoginLayout({ updateUser, updatePassword, logInSuccessful }) {
+  return (
+    <ImageBackground style={styles.loginContainer} source={backgroundImage}>
+      <TextInput
+        style={[styles.textInput, styles.container]}
+        placeholder={USER_PLACE_HOLDER}
+        textAlign="center"
+        onChangeText={text => updateUser(text)}
+      />
+      <TextInput
```
Comment (santinucera) [inline WBooks/src/app/screens/Login/layout.js]: This `TextInout` is pretty similar to the upper one, try to abstract the repeated logic

File: WBooks/src/app/screens/BookDetail/components/DetailBox/index.js [hunk 4]
```diff
@@ -23,45 +21,18 @@ class DetailBox extends Component {
...
-          <TouchableOpacity
-            onPress={this.handleRentBook}
-            style={[styles.genericStyleButton, rentButtonStyle]}
-            disabled={disabled}
-          >
-            <Text style={[styles.genericTextButton, styles.textAvailable]}>{buttonText.rent}</Text>
-          </TouchableOpacity>
-        </View>
-      </View>
+      <DetailBoxLayout
```
Comment (santinucera) [inline WBooks/src/app/screens/BookDetail/components/DetailBox/index.js]: I will move to the layout file all `availabilityStyle`, `availabilityText `,  `rentButtonStyle ` and `disabled `

File: WBooks/src/app/screens/Library/components/BookList/layout.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React, { Fragment } from 'react';
+import { FlatList } from 'react-native';
+
+function BookListLayout({ data, keyExtractor, renderItem, ItemSeparatorComponent }) {
+  return (
+    <Fragment>
```
Comment (santinucera) [inline WBooks/src/app/screens/Library/components/BookList/layout.js]: I think this `Fragment` is not necessary

File: WBooks/src/app/screens/Login/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+class Login extends Component {
+  state = {
+    user: '',
+    password: ''
+  };
+
+  updateUser = text => this.setState({ user: text });
+
+  updatePassword = text => this.setState({ password: text });
```
Comment (santinucera) [inline WBooks/src/app/screens/Login/index.js]: This function is pretty similar to the upper one

File: WBooks/src/app/screens/BookDetail/index.js (hunk 2 above)
Comment (felire) [reply WBooks/src/app/screens/BookDetail/index.js]: Yes, if i remove this View, the commentBox bottom won't see okey. I tried other options but they haven't worked

File: WBooks/src/app/screens/Login/layout.js (hunk 3 above)
Comment (felire) [reply WBooks/src/app/screens/Login/layout.js]: Do you think that i should make a component?

Comment (felire) [reply WBooks/src/app/screens/Login/layout.js]: I create a function to abstract the logic of the updates function, i made a partial function. it's something like this: update = param => text => this.setState({ [param]: text }); 
And it's used to the tupdatePassword and user, doing: updatePassword = update('password'), and then text => updatePassword(text). With the curry library is more simple to make partial functions.

File: WBooks/src/app/screens/BookDetail/components/DetailBox/index.js (hunk 4 above)
Comment (felire) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/index.js]: But that props are logic. I thought that the layout show something and it don't  have logic. Or when the logic is to decide what to show it get into the layout file?

**Pull request merged**